catroid/src/androidTestPhiro/java/org/catrobat/catroid/uiespresso/ui/activity/PhiroSettingsActivityTest.java

```diff
@@ -0,0 +1,151 @@
+			BaseActivityInstrumentationRule<>(SettingsActivity.class, true, false);
+
+	private List<String> allSettings = new ArrayList<>(Arrays.asList(SETTINGS_SHOW_PHIRO_BRICKS, PHIRO_INITIALIZED));
+	private Map<String, Boolean> initialSettings = new HashMap<>();
+
+	@Before
+	public void setUp() {
+		SharedPreferences sharedPreferences = PreferenceManager
+				.getDefaultSharedPreferences(InstrumentationRegistry.getTargetContext());
+
+		for (String setting : allSettings) {
+			initialSettings.put(setting, sharedPreferences.getBoolean(setting, false));
+		}
+		setAllSettingsTo(false);
+
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@Test
+	public void phiroIsInitiallyActivatedTest() {
+		SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(InstrumentationRegistry.getTargetContext());
+		assertTrue("Phiro preference was not initially enabled!", sharedPreferences.getBoolean(SETTINGS_SHOW_PHIRO_BRICKS, false));
+	}
+
+	@Test
+	public void phiroIsFirstVisibleSettingTest() {
+		onData(PreferenceMatchers.withTitle(R.string.preference_title_enable_phiro_bricks))
+				.atPosition(0)
+				.check(matches(isDisplayed()));
+	}
+
+	@Test
+	public void phiroSettingsTest() {
+		onData(PreferenceMatchers.withTitle(R.string.preference_title_enable_phiro_bricks))
+				.perform(click());
+
+		checkPreference(R.string.preference_title_enable_phiro_bricks, SETTINGS_SHOW_PHIRO_BRICKS);
+
+		String phiroLink = UiTestUtils.getResourcesString(R.string.phiro_preference_link);
+
+		Intents.init();
+		Matcher<Intent> expectedIntent = allOf(hasAction(Intent.ACTION_VIEW), hasData(phiroLink));
+		intending(expectedIntent).respondWith(new Instrumentation.ActivityResult(0, null));
+		onData(PreferenceMatchers.withTitle(R.string.phiro_preference_title)).perform(click());
+		intended(expectedIntent);
+		Intents.release();
+	}
+
+	@After
+	public void tearDown() {
+		SharedPreferences.Editor sharedPreferencesEditor = PreferenceManager
+				.getDefaultSharedPreferences(InstrumentationRegistry.getTargetContext()).edit();
+		for (String setting : initialSettings.keySet()) {
+			sharedPreferencesEditor.putBoolean(setting, initialSettings.get(setting));
+		}
+		sharedPreferencesEditor.commit();
+		initialSettings.clear();
+	}
+
+	private void setAllSettingsTo(boolean value) {
+		SharedPreferences.Editor sharedPreferencesEditor = PreferenceManager
+				.getDefaultSharedPreferences(InstrumentationRegistry.getTargetContext()).edit();
+
+		for (String setting : allSettings) {
+			sharedPreferencesEditor.putBoolean(setting, value);
+		}
+		sharedPreferencesEditor.commit();
+	}
+
+	private void checkPreference(int displayedTitleResourceString, String sharedPreferenceTag) {
+		SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(InstrumentationRegistry.getTargetContext());
+
+		onData(PreferenceMatchers.withTitle(displayedTitleResourceString)).perform(click());
```

Comment:
please reformat longer espresso calls like this to format:
onWhatever(.......)
     .doWhatever(....)

eg. 
onData(PreferenceMatchers.withTitle(displayedTitleResourceString))
      .perform(click());